### PR TITLE
ES consume migration of LumiAlCaRecoProducer

### PR DESCRIPTION
**Pull request by tvami**
#### PR description:

ES consume migration of LumiAlCaRecoProducer

#### PR validation:

I tried to run the unit tests but got file not found error :( 
Maybe it will work for Jenkins
Anyway the code compiles

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

N/A

### CODE REVIEW

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
OK unit test is running fine: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-5ef42c/20475/unitTests/src/Calibration/LumiAlCaRecoProducers/test/TestCalibrationLumiAlCaRecoProducers/testing.log

**Comment by francescobrivio** [comment]
+alca

 - part of ES migration campaign 
 - jenkins tests pass nicely

**Comment by qliphy** [comment]
+1

**Comment by qliphy** [comment]
@tvami There is a IB issue appearing after merging this PR, which seems to be related:
https://cmssdt.cern.ch/SDT/html/cmssdt-ib/#/relVal/CMSSW_12_2/2021-11-12-2300?selectedArchs=slc7_amd64_gcc900&selectedFlavors=X&selectedStatus=failed

eos/cms/store/user/cmsbuild/store/data/Run2016H/AlCaLumiPixels1/RAW/v1/000/283/383/00000/BAF61397-DA95-E611-B879-02163E0145CC.root
----- Begin Fatal Exception 13-Nov-2021 07:23:03 CET-----------------------
An exception of category 'ESGetTokenWrongTransition' occurred while
   [0] Processing global end LuminosityBlock run: 283383 luminosityBlock: 1
   [1] Calling method for module RawPCCProducer/'ALCARECORawPCCProd'
Exception Message:
The transition ID stored in the ESGetToken does not match the
transition where the token is being used. The associated record
type is: LumiCorrectionsRcd

Would you please have a check?

File: Calibration/LumiAlCaRecoProducers/plugins/RawPCCProducer.cc [hunk 1]
```diff
@@ -55,7 +56,8 @@ class RawPCCProducer : public edm::global::EDProducer<edm::EndLuminosityBlockPro
 
 //--------------------------------------------------------------------------------------------------
 RawPCCProducer::RawPCCProducer(const edm::ParameterSet& iConfig)
-    : putToken_{produces<LumiInfo, edm::Transition::EndLuminosityBlock>(
+    : lumiCorrectionsToken_(esConsumes()),
```
Comment (Dr15Jones) [inline Calibration/LumiAlCaRecoProducers/plugins/RawPCCProducer.cc:59]: This needs to state it will be used at the end LuminosityBlock transition.

Comment (tvami) [reply Calibration/LumiAlCaRecoProducers/plugins/RawPCCProducer.cc:59]: like this @Dr15Jones ?
```suggestion
    : lumiCorrectionsToken_(esConsumes<edm::Transition:: beginLuminonosityBlock >(),
```

Comment (tvami) [reply Calibration/LumiAlCaRecoProducers/plugins/RawPCCProducer.cc:59]: ok `1020.0` is running like this, so I'll submit a PR in a second. Interesting that this wasnt picked up by the unit test, I thought https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-5ef42c/20475/unitTests/src/Calibration/LumiAlCaRecoProducers/test/TestCalibrationLumiAlCaRecoProducers/ is running this

**Pull request merged**